libraries/AP_MSP/msp.cpp

```diff
@@ -177,7 +177,7 @@ bool MSP::msp_parse_received_data(msp_port_t *msp, uint8_t c)
         break;
 
     case MSP_HEADER_M:      // Waiting for '<'
-        if (c == '<') {
+        if ((c == '<')|| (c=='>')) {
```

Comment:
This indicates a packet in the opposite direction and appears wrong to me. If correct the comment needs fixing as well

Reply:
On the HDZero the direction was opposite, and the parser failed.

Reply:
It makes some sense, but also seems counter to what the function is trying to do and it's name - might be worth checking betaflight/src/main/msp/msp_serial.c from which this was ported to see if that code has evolved in a compatible way